metadata-ingestion/tests/unit/sdk_v2/test_lineage_client.py

```diff
@@ -561,3 +562,261 @@ def test_add_datajob_lineage_validation(client: DataHubClient) -> None:
+    )
+
+    downstream_schema = SchemaMetadataClass(
+        schemaName="downstream_table",
+        platform="urn:li:dataPlatform:snowflake",
+        version=1,
+        hash="1234567890",
+        platformSchema=OtherSchemaClass(rawSchema=""),
+        fields=[
+            SchemaFieldClass(
+                fieldPath="id",
+                type=SchemaFieldDataTypeClass(type=StringTypeClass()),
+                nativeDataType="string",
+            ),
+        ],
+    )
+    # Test case 3: Auto strict column lineage
+    with patch.object(LineageClient, "_get_fields_from_dataset_urn") as mock_method:
+        # Configure the mock with a simpler side effect function
+        mock_method.side_effect = lambda urn: sorted(
+            {
+                field.fieldPath
+                for field in (
+                    upstream_schema if "upstream" in str(urn) else downstream_schema
+                ).fields
+            }
+        )
+
+        # Run the lineage function
+        client.lineage.add_dataset_copy_lineage(
+            upstream=upstream,
+            downstream=downstream,
+            column_lineage="auto_strict",
+        )
+
+    assert_client_golden(
+        client, _GOLDEN_DIR / "test_lineage_copy_auto_strict_golden.json"
+    )
+
+
+def test_add_lineage_dataset_to_dataset_copy_custom_mapping(
+    client: DataHubClient,
+) -> None:
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,upstream_table,PROD)"
+    downstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,downstream_table,PROD)"
+
+    # Test case 4: Custom column mapping
+    column_mapping = {"Age": ["name", "customer_id"]}
+    client.lineage.add_lineage(
+        upstream=upstream, downstream=downstream, column_lineage=column_mapping
+    )
+    assert_client_golden(
+        client, _GOLDEN_DIR / "test_lineage_custom_mapping_golden.json"
+    )
+
+
+def test_add_lineage_dataset_to_dataset_transform(client: DataHubClient) -> None:
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,upstream_table,PROD)"
+    downstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,downstream_table,PROD)"
+    query_text = "SELECT user_id as userId, full_name as fullName FROM upstream_table"
+    column_mapping = {"userId": ["user_id"], "fullName": ["full_name"]}
+    # Test case 5: Dataset to dataset with transformation and column mapping
+    client.lineage.add_lineage(
+        upstream=upstream,
+        downstream=downstream,
+        query_text=query_text,
+        column_lineage=column_mapping,
+    )
+    assert_client_golden(client, _GOLDEN_DIR / "test_lineage_transform_golden.json")
+
+
+def test_add_lineage_dataset_to_datajob(client: DataHubClient) -> None:
+    """Test add_lineage method with dataset to datajob."""
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,source_table,PROD)"
+    downstream = (
+        "urn:li:dataJob:(urn:li:dataFlow:(airflow,example_dag,PROD),process_job)"
+    )
+
+    # Call the method
```

Comment:
useless comment

in general, this file feels like it's good a couple too many comments